src/courtroom.cpp

```diff
@@ -2591,101 +2607,86 @@ void Courtroom::append_ic_text(QString p_text, QString p_name, QString p_action,
   QTextCharFormat bold;
   QTextCharFormat normal;
   QTextCharFormat italics;
+  QTextBlockFormat format;
   bold.setFontWeight(QFont::Bold);
   normal.setFontWeight(QFont::Normal);
   italics.setFontItalic(true);
+  format.setTopMargin(log_margin);
   const QTextCursor old_cursor = ui_ic_chatlog->textCursor();
   const int old_scrollbar_value = ui_ic_chatlog->verticalScrollBar()->value();
-
-  if (p_action == "")
-    p_text = filter_ic_text(p_text, ao_app->is_colorlog_enabled(), -1, color);
-
-  if (log_goes_downwards) {
-    const bool is_scrolled_down =
-        old_scrollbar_value == ui_ic_chatlog->verticalScrollBar()->maximum();
-
-    ui_ic_chatlog->moveCursor(QTextCursor::End);
-
-    if (!first_message_sent) {
-      ui_ic_chatlog->textCursor().insertText(p_name, bold);
-      first_message_sent = true;
-    }
-    else {
-      ui_ic_chatlog->textCursor().insertText('\n' + p_name, bold);
-    }
-
-    if (p_action != "") {
-      ui_ic_chatlog->textCursor().insertText(" " + p_action + ": ", normal);
-      ui_ic_chatlog->textCursor().insertText(p_text + ".", italics);
-    }
-    else {
+  const bool need_newline = !ui_ic_chatlog->document()->isEmpty();
+  const int scrollbar_target_value = log_goes_downwards ? ui_ic_chatlog->verticalScrollBar()->maximum() : ui_ic_chatlog->verticalScrollBar()->minimum();
+
+  ui_ic_chatlog->moveCursor(log_goes_downwards ? QTextCursor::End : QTextCursor::Start);
+
+  // Only prepend with newline if log goes downwards
+  if (log_goes_downwards && need_newline) {
+    ui_ic_chatlog->textCursor().insertBlock(format);
+  }
+
+  // Timestamp if we're doing that meme
+  if (log_timestamp)
+    ui_ic_chatlog->textCursor().insertText("[" + QDateTime::currentDateTime().toUTC().toString("h:mm:ss AP") + "] ", normal);
```

Comment:
Why `toUTC()`?

Reply:
Because the client's text logs are UTC as well, I can strip it out tho cuz I hate it